### PR TITLE
Update to egui 0.17

### LINKED ISSUE #56: update to egui 0.17

### PR DESCRIPTION
**Pull request by bobtwinkles**
I'm not entirely sure that the sampler injection (`egui_texture_from_wgpu_texture` and friends) works, I haven't tested it yet.

Fixes #56

### CODE REVIEW

**Comment by asayers** [comment]
I tested this branch; it seems to work.  (Didn't test user textures.)

**Comment by hasenbanck** [comment]
Thank you for the PR. Some comments I have:

1. You pulled in the log crate as dependency for something that seems like debug output. egui itself also optionaly uses the tracing crate. Please remove the dependency or make it also optional based on tracing so that it aligned with the upstream project.
2. Please don't use unwrap(). If you think it can fail, than please return an error. You also seem to test if the texture sizes are bigger than an u32, which is fine, but then it should return a custom error instead of panicing. I also don't think that this error can be practically be thrown, since WGPU will most likely throw a "Error::Validation" before, since device limits are much lower anyhow.

**Comment by bobtwinkles** [comment]
> You pulled in the log crate as dependency for something that seems like debug output.

Removed. I added it this way to align with `wgpu` conventions (which exposes detailed information via `log`).

>  Please don't use unwrap().

I don't expect most of these to get hit in practice either, so I've ripped them out.

File: src/lib.rs [hunk 1]
```diff
@@ -360,131 +362,143 @@ impl RenderPass {
...
+                egui::ImageData::Color(_) => None,
+                egui::ImageData::Alpha(a) => Some(a.srgba_pixels(1.0).collect()),
+            };
+
+            let image_data: &[u8] = match &image_delta.image {
+                egui::ImageData::Color(c) => bytemuck::cast_slice(c.pixels.as_slice()),
+                egui::ImageData::Alpha(_) => {
+                    // The unwrap here should never fail as alpha_srgb_pixels will have been set to
+                    // `Some` above.
+                    bytemuck::cast_slice(alpha_srgb_pixels.as_ref().unwrap().as_slice())
```
Comment (hasenbanck) [inline src/lib.rs:402]: Even if it's impossible in the current state of the code, don't throw a generic panic and at least specify the panic reason  with "expect()".

**Comment by hasenbanck** [comment]
Looks fine now. I will give people time to test the changes and then release a now version to cargo in some days.

**Comment by bobtwinkles** [comment]
Thanks for the review here and on the `winit` integration, take care!

**Pull request merged**